bug: assignment to entry in nil map when --watch-namespaces flag is provided

### Describe the issue

When providing the --watch-namespaces the operator fails to boot with the following error:

```
panic: assignment to entry in nil map

goroutine 1 [running]:
main.main()
	/workspace/main.go:109 +0x15e8
```

### To Reproduce

Start the operator with the `--watch-namespaces` flag set.

### Expected behavior

It should not panic

### Your Environment

```markdown
- Fluent Operator version: 3.1.0
- Container Runtime:
- Operating system:
- Kernel version:
```


### How did you install fluent operator?

Helm chart

### Additional context

_No response_

--- a/cmd/fluent-manager/main.go
+++ b/cmd/fluent-manager/main.go
@@ -17,11 +17,10 @@ limitations under the License.
 package main
 
 import (
+	"errors"
 	"flag"
 	"os"
 	"strings"
-
-	"errors"
 
 	"github.com/joho/godotenv"
 	"sigs.k8s.io/controller-runtime/pkg/cache"
@@ -105,6 +104,8 @@ func main() {
 		config := cache.Config{}
 		namespacedController = true
 		namespaces := strings.Split(watchNamespaces, ",")
+
+		ctrlOpts.Cache.DefaultNamespaces = make(map[string]cache.Config)
 		for _, namespace := range namespaces {
 			ctrlOpts.Cache.DefaultNamespaces[namespace] = config
 		}

